Сообщение коммита

OurnoFrontendDev opened the pull request.

OurnoFrontendDev committed:
Сообщение коммита

danimaxi54 commented on src/App.tsx:
```diff
@@ -0,0 +1,30 @@
+import {AppWrapper, HeaderContainer} from "./AppStyles";
+
+function App() {
+    const theme = useSelector((state: RootState) => state.theme?.theme);
```
нужно создать useTypedSelector, чтобы каждый раз не типизировать state: RootState

OurnoFrontendDev replied:
Исправил

OurnoFrontendDev replied:
Сделал

danimaxi54 commented on src/App.tsx:
```diff
@@ -0,0 +1,30 @@
+function App() {
+    const theme = useSelector((state: RootState) => state.theme?.theme);
+    return (<>
+            <ThemeProvider theme={theme === 'dark' ? darkTheme : lightTheme}>
```
ThemeProdiver уже должен знать о теме, ты не должен прокидывать туда тему

danimaxi54 commented on src/App.tsx:
```diff
@@ -0,0 +1,30 @@
+                <FontStyles/>
+                <GlobalStyle/>
+                <AppWrapper>
+                    <HeaderContainer>
```
не нужен здесь HeaderContainer 
помести сразу в SearchTogglePanelContainer

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/AppStyles.ts:
```diff
@@ -0,0 +1,22 @@
+`;
+export const HeaderContainer = styled.header`
+    width: 100%;
+    height: 5.6%;
```
откажись от процентов и сделай пикселями

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/Days/Day'sStyledCompmonents.ts:
```diff
@@ -0,0 +1,62 @@
+    height: 181px;
+    border-radius: 22px;
+    border: 1px solid #E8EAF6;
+    background-color: ${({ theme }) => theme.backgroundForecast || '#000'};
```
тему ты не должен прокидывать
нужно сделать на css переменных всю тему

danimaxi54 commented on src/components/Days/Day'sStyledCompmonents.ts:
```diff
@@ -0,0 +1,62 @@
+
+export const ScrollArea = styled.div`
+    display: flex;
+    transition: transform 0.3s ease; /* Smooth scroll */
```
/* Smooth scroll */ коммент не нужен

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/Days/Day'sStyledCompmonents.ts:
```diff
@@ -0,0 +1,62 @@
+    display: inline-block;
+    margin-left: 10px;
+    text-align: center;
+    border: 1px solid #eee;
```
#eee нужно цвета заносить в css переменные

danimaxi54 commented on src/components/Days/Day'sStyledCompmonents.ts:
```diff
@@ -0,0 +1,62 @@
+    border-radius: 8px;
+    width: 86px;
+    height: 97px;
+    background-color: ${({ theme }) => theme.backgroundColor};
```
все цвета должны быть в css переменных

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,90 @@
+    ForecastItemContainer,
+    ForecastTextContainer,
+    ForecastWrapper, ScrollArea
+} from "./Day'sStyledCompmonents";
```
./Day'sStyledCompmonents ошибка в нейминге

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,90 @@
+        const ampm = hours >= 12 ? 'PM' : 'AM';
+        hours = hours % 12;
+        hours = hours ? hours : 12;
+        const minutesStr = minutes < 10 ? `0${minutes}` : minutes;
```
return отбей пустой линией сверху

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,90 @@
+    };
+
+    const filteredData = getFilteredHourlyData();
+    filteredData.sort((a, b) => a.dt - b.dt);
```
в useMemo закутай сортировку. и можно в одну строку написать getFilteredHourlyData().sort((a, b) => a.dt - b.dt);

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,90 @@
+
+    const filteredData = getFilteredHourlyData();
+    filteredData.sort((a, b) => a.dt - b.dt);
+    const maxScroll = (filteredData.length * 110) - 400;
```
400 нужно вынести в отдельную переменную 
не понятно, что за 400 такие

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,90 @@
+    filteredData.sort((a, b) => a.dt - b.dt);
+    const maxScroll = (filteredData.length * 110) - 400;
+
+    const scrollStep = 100;
```
вынести за компонент

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,90 @@
+        return <img src={iconUrl} alt="Weather Icon" />;
+    };
+
+    const arrMap = filteredData.map((hour, index) => (
```
arrMap не понятное название переменной

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,90 @@
+
+    const WeatherIcon: React.FC<{ iconCode: string }> = ({ iconCode }) => {
+        const iconUrl = `https://openweathermap.org/img/w/${iconCode}.png`;
+        return <img src={iconUrl} alt="Weather Icon" />;
```
return надо отбить пустой линией сверху

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/SearchTogglePanel/SearchTogglePanelStyles.ts:
```diff
@@ -0,0 +1,22 @@
+import styled from "styled-components";
+import {DefaultFlexSettings} from "../../reusedStyleVariables";
+
+export const CityFinderTemperatureSwitchThemeSwitchBarContainer = styled.section`
```
CityFinderTemperatureSwitchThemeSwitchBarContainer сложное название, упрощай

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/TempUnitSwitcher/index.tsx:
```diff
@@ -0,0 +1,27 @@
+
+export const TempUnitSwitcher = () => {
+    const dispatch = useDispatch();
+    const unit = useSelector((state: any) => state.weather.unit);
```
state: any нельзя так 
useTypedSelector нужен

danimaxi54 commented on src/components/TempUnitSwitcher/tempUnitSwitcherStyles.ts:
```diff
@@ -0,0 +1,46 @@
+    left: ${(props) => (props.checked ? 'calc(100% - 1.75rem - 2px)' : '1px')};
+    height: 1.75rem; /* 7 * 0.25rem */
+    width: 1.75rem;
+    background-color: #4f46e5; /* bg-indigo-400 */
```
убрать комменты везде

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/ThisDay/ThisDayStyles.ts:
```diff
@@ -0,0 +1,63 @@
+    }
+`;
+
+export const LocationDescriptionElementContainer = styled.div`
```
Element слово можно не использовать

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/ThisDay/ThisDayStyles.ts:
```diff
@@ -0,0 +1,63 @@
+    color: ${({ theme }) => theme.text};
+    justify-content: flex-start;
+    align-items: center;
+    padding: 2.1% 0 0 3.6%;
```
проценты убрать, на px перейти

danimaxi54 commented on src/components/ThisDay/index.tsx:
```diff
@@ -0,0 +1,60 @@
+} from "./ThisDayStyles";
+
+export const ThisDay = () => {
+    const dispatch = useDispatch<AppDispatch>();
```
useDispatch<AppDispatch>();
 а зачем здесь затипизировал?

OurnoFrontendDev replied:
исправил

danimaxi54 commented on src/components/ThisDay/index.tsx:
```diff
@@ -0,0 +1,60 @@
+    function WeatherIcon() {
+        const iconUrl = `https://openweathermap.org/img/w/${weather?.currentWeather?.weather[0].icon}.png`;
+        return <img src={iconUrl} alt="Weather Icon"/>;
+    }
```
отбить пустой линией снизу

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/WeatherUnionInfo/WeatherUnionInfoStyles.ts:
```diff
@@ -0,0 +1,45 @@
+
+export const WeatherUnionInfoContainer = styled.section`
+    ${DefaultFlexSettings};
+    gap: 2%;
```
убрать проценты

danimaxi54 commented on src/components/WeatherUnionInfo/WeatherUnionInfoStyles.ts:
```diff
@@ -0,0 +1,45 @@
+        margin-top: 6%;
+    } 
+    @media screen and (max-width: 800px) {
+        visibility: hidden;
```
а зачем скрываешь?

danimaxi54 commented on src/components/WeatherUnionInfo/WeatherUnionInfoStyles.ts:
```diff
@@ -0,0 +1,45 @@
+    flex-direction: column;
+    width: 100%;
+    height: 75%;
+    margin-top: 1%;
```
на px перевести!

danimaxi54 commented on src/components/WeatherUnionInfo/index.tsx:
```diff
@@ -0,0 +1,29 @@
+    return (
+        <WeatherUnionInfoContainer>
+            <WeatherTownInfoContainer>
+                <ThisDay></ThisDay>
```
<ThisDay></ThisDay> => <ThisDay />

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/WeatherUnionInfo/index.tsx:
```diff
@@ -0,0 +1,29 @@
+                </ForeCastContainer>
+            </WeatherTownInfoContainer>
+            <WeekForecastContainer>
+                <WeekForecast>
```
пройдись по компоненту
закрывающие теги не нужны, так как нет children

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/cityWeatherSearch/index.tsx:
```diff
@@ -0,0 +1,51 @@
+import {fetchWeatherBySearchRequest} from "../features/weatherSlice";
+import styled from 'styled-components';
+import {fonts} from "../../fonts/fontStyles";
+
```
лишняя пустая линия

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/cityWeatherSearch/index.tsx:
```diff
@@ -0,0 +1,51 @@
+
+
+const InputSearchBar = styled.input.attrs({
+
```
лишня пустая линия

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/cityWeatherSearch/index.tsx:
```diff
@@ -0,0 +1,51 @@
+
+const InputSearchBar = styled.input.attrs({
+
+    placeholder: "Search for cities",
```
Placholder сверху должен прокинуть, а не здесь

danimaxi54 commented on src/components/cityWeatherSearch/index.tsx:
```diff
@@ -0,0 +1,51 @@
+    };
+
+    return (
+        <>
```
не нужен тут фрагмент

OurnoFrontendDev replied:
Исправил

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/cityWeatherSearch/index.tsx:
```diff
@@ -0,0 +1,51 @@
+    const weather = useSelector((state: any) => state.weather);
+
+    const handleSearch = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (city.trim() !== '' && e.key === "Enter") {
```
city.trim() вот так достаточно написать

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/features/themeSlice.ts:
```diff
@@ -0,0 +1,27 @@
+    theme: 'light',
+};
+
+const themeSlice = createSlice({
```
не нужен стор для темы на будущее 
ThemeProvider достаточно создать

danimaxi54 commented on src/components/features/weatherSaga.ts:
```diff
@@ -0,0 +1,79 @@
+import {RootState} from "../../state/store";
+import {WeatherResponse} from "../../types/stateType";
+
+export const API_KEY = 'fe4feefa8543e06d4f3c66d92c61b69c';
```
ключ и url должны лежать в env файлике

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/svg/SvgLoader.tsx:
```diff
@@ -0,0 +1,31 @@
+
+interface IconClickableProps extends IconBaseProps {
+    Svg: React.VFC<React.SVGProps<SVGSVGElement>>;
+}
```
снизу пустую линию поставь
тип прилип к компоненту

anton-novikov997 replied:
исправил

danimaxi54 commented on src/components/weatherDetails/WeatherDetailsStyles.ts:
```diff
@@ -0,0 +1,96 @@
+    width: 100%;
+    height: 23%;
+
+`
```
пустая линия лишняя

anton-novikov997 replied:
исправил

danimaxi54 commented on src/components/weatherDetails/index.tsx:
```diff
@@ -0,0 +1,70 @@
+    const hourlyData = useSelector((state: RootState) => state.weather || {});
+
+    const formatTime = (timestamp: number): string => {
+        const date = new Date(timestamp * 1000); // Convert from seconds to milliseconds
```
// Convert from seconds to milliseconds не нужны комменты

anton-novikov997 replied:
исправил

danimaxi54 commented on src/components/weekForecast/index.tsx:
```diff
@@ -0,0 +1,65 @@
+} from './weekForecastStyles';
+
+export const WeekForecast = () => {
+    const dailyForecasts = useSelector((state: RootState) => state.weather.forecasts?.daily || []);
```
state.weather.forecasts?.daily || [] каждый раз у тебя будет перерендре лишний компонента возникать, так как ты в useSelector новую сслылку будешь отдавать

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on src/components/weekForecast/index.tsx:
```diff
@@ -0,0 +1,65 @@
+        const timestamp = forecast.dt * 1000;
+        const dateObject = new Date(timestamp);
+        const formattedDate = format(dateObject, 'd MMM, EEEE');
+        return {
```
отбей return линией сверху

anton-novikov997 replied:
исправил

danimaxi54 commented on src/fonts/OFL.txt:
```diff
@@ -0,0 +1,93 @@
+Copyright 2014 The Nunito Project Authors (https://github.com/googlefonts/nunito)
+
```
а этот файлик зачем?

anton-novikov997 replied:
исправил

danimaxi54 commented on src/fonts/README.txt:
```diff
@@ -0,0 +1,79 @@
+
+This download contains Nunito as both variable fonts and static fonts.
+
+Nunito is a variable font with this axis:
```
readme не нужен

anton-novikov997 replied:
исправил

danimaxi54 commented on src/fonts/fontStyles.ts:
```diff
@@ -0,0 +1,45 @@
+import NunitoRegular from "./Nunito-Regular.ttf";
+import NunitoSemiBold from "./Nunito-SemiBold.ttf";
+
+
```
пустые линии лишние 
не нужны

anton-novikov997 replied:
исправил

danimaxi54 commented on src/index.tsx:
```diff
@@ -0,0 +1,35 @@
+    }
+`;
+
+import React from 'react';
```
import сверху должен быть

anton-novikov997 replied:
исправил

danimaxi54 commented on src/reportWebVitals.ts:
```diff
@@ -0,0 +1,15 @@
+import { ReportHandler } from 'web-vitals';
+
+const reportWebVitals = (onPerfEntry?: ReportHandler) => {
```
не нужен этот файлик

anton-novikov997 replied:
исправил

danimaxi54 commented on src/themes.ts:
```diff
@@ -0,0 +1,23 @@
+
+export const lightTheme :Theme = {
+    color: '#000',
+    toggleBorder: '#FFF',
```
css переменные нужны

danimaxi54 commented on src/types/stateType.ts:
```diff
@@ -0,0 +1,95 @@
+export type Temperature = {
```
weather.ts файлик назови

что за stateType

anton-novikov997 replied:
исправил

danimaxi54 commented on src/types/stateType.ts:
```diff
@@ -0,0 +1,95 @@
+    };
+    unit: 'metric' | 'imperial';
+}
+export const initialState: WeatherState = {
```
а что в файлике с типизацией делает константа для хранилища??

anton-novikov997 replied:
исправил

danimaxi54 commented on src/types/stateType.ts:
```diff
@@ -0,0 +1,95 @@
+export type WeatherState = {
+    loading: boolean;
+    error: string | null;
+    currentWeather: any;
```
почему any
нельзя

OurnoFrontendDev replied:
Исправил

danimaxi54 commented on webpack.config.ts:
```diff
@@ -0,0 +1,226 @@
+//             new MiniCssExtractPlugin(),
+//             new CleanWebpackPlugin(), // Переместите сюда
+//         ]
+//     };
```
чему куча закоменченного кода?

anton-novikov997 replied:
исправил

OurnoFrontendDev committed:
Внес правки
Переделал тему
Поменял цвета
Добавил адаптив
Добавил функционал поиска по городам

OurnoFrontendDev committed:
Переделал адптив

danimaxi54 commented on webpack.config.ts:
```diff
@@ -0,0 +1,114 @@
+                template: './public/index.html'
+            }),
+            new MiniCssExtractPlugin(),
+            new CleanWebpackPlugin(), // Переместите сюда
```
лишний коммент

danimaxi54 commented on src/themes.ts:
```diff
@@ -0,0 +1,40 @@
+    '--input-background-color': string
+    '--arrow-color': string
+    'input-background-color': string
+
```
лишние пустые линии

danimaxi54 commented on src/skeleton/Skeleton.tsx:
```diff
@@ -0,0 +1,45 @@
+    }
+`;
+
+export const Skeleton: React.FC<{ width?: string; height?: string }> = ({ width = '100%', height = '100%' }) => {
```
type Props создай и туда занеси типизацию

danimaxi54 commented on src/provider/index.tsx:
```diff
@@ -0,0 +1,68 @@
+}
+
+export const ThemeProvider: React.FC<ThemeProviderProps> = ({children}) => {
+    const [theme, setTheme] = useState<string>('lightTheme');
```
в enum положи тему

danimaxi54 commented on src/provider/index.tsx:
```diff
@@ -0,0 +1,68 @@
+
+    return (
+        <ThemeContext.Provider value={contextValue}>
+            <StyledThemeProvider theme={currentTheme}>
```
это действительно нужно?

danimaxi54 commented on src/provider/index.tsx:
```diff
@@ -0,0 +1,68 @@
+
+    const currentTheme = theme === 'lightTheme' ? lightTheme : darkTheme;
+
+    const contextValue: ThemeContextType = {
```
в useMemo оберни объект. у тебя могут быть перерендеры ThemeProvider, поэтому лишний раз компоненты будут перерндериваться из-за того, что просто изменилась ссылка на объект внутри контекста

danimaxi54 commented on src/provider/index.tsx:
```diff
@@ -0,0 +1,68 @@
+                mediaQuery.removeEventListener('change', handleChange);
+            };
+        }
+    }, [manualOverride]);
```
отступ снизу сделай

danimaxi54 commented on src/provider/index.tsx:
```diff
@@ -0,0 +1,68 @@
+            };
+        }
+    }, [manualOverride]);
+    const changeTheme = (newTheme: string) => {
```
newTheme: ThemeEnum, а не newTheme: string

danimaxi54 commented on src/provider/index.tsx:
```diff
@@ -0,0 +1,68 @@
+    const changeTheme = (newTheme: string) => {
+        setTheme(newTheme);
+        setManualOverride(true);
+        localStorage.setItem('theme', newTheme);
```
создать константы для значений, которые сетишь в ls - THEME_LOCAL_STORAGE_KEY И т. д

danimaxi54 commented on src/provider/index.tsx:
```diff
@@ -0,0 +1,68 @@
+        localStorage.setItem('manualOverride', 'true');
+    };
+
+    const currentTheme = theme === 'lightTheme' ? lightTheme : darkTheme;
```
зачем эта проверка, если у тебя на 11 строке есть theme

danimaxi54 commented on src/provider/index.tsx:
```diff
@@ -0,0 +1,68 @@
+
+    useEffect(() => {
+        const storedTheme = localStorage.getItem('theme');
+        const storedManualOverride = localStorage.getItem('manualOverride') === 'true';
```
зачем это?
первый делом проверяешь есть ли theme в ls. есть есть, то ставишь ее
а иначе тему системы смотришь

danimaxi54 commented on src/provider/index.tsx:
```diff
@@ -0,0 +1,68 @@
+
+    const [manualOverride, setManualOverride] = useState<boolean>(false);
+
+    useEffect(() => {
```
слишком сложнй useEffect, проще делай

danimaxi54 commented on src/hooks/useTheme.ts:
```diff
@@ -0,0 +1,10 @@
+
+export const useTheme = (): ThemeContextType => {
+    const context = useContext(ThemeContext);
+    if (!context) {
```
у тебя контекст всегда будет

danimaxi54 commented on src/env.ts:
```diff
@@ -0,0 +1,2 @@
+export const API_KEY = 'fe4feefa8543e06d4f3c66d92c61b69c';
```
в env файлик положи такое

danimaxi54 commented on src/env.ts:
```diff
@@ -0,0 +1,2 @@
+export const API_KEY = 'fe4feefa8543e06d4f3c66d92c61b69c';
+export const BASE_URL = 'https://api.openweathermap.org/data/2.5';
```
в env

danimaxi54 commented on src/components/weekForecast/weekForecastStyles.ts:
```diff
@@ -0,0 +1,107 @@
+    font-family: ${fonts.Body};
+    padding-top: 10px;
+    padding-left: 19px;
+    color: ${({theme}) => theme['--card-title-color']};
```
в будущем тебе нужно не theme принимать, а css переменные делать, так как ты щас везде theme передаешь

danimaxi54 commented on src/components/weekForecast/index.tsx:
```diff
@@ -0,0 +1,71 @@
+import {useTypedSelector} from "../../hooks/useTypedSelector";
+
+export const WeekForecast = () => {
+
```
пустая линия лишняя

danimaxi54 commented on src/components/weatherDetails/index.tsx:
```diff
@@ -0,0 +1,115 @@
+        if (weatherDetails.length > 0) {
+            setIsLoading(false);
+        }
+    }, [weatherDetails]);
```
отбей пустой линией снизу

danimaxi54 commented on src/components/weatherDetails/index.tsx:
```diff
@@ -0,0 +1,115 @@
+    }, [hourlyData]);
+
+    useEffect(() => {
+        if (weatherDetails.length > 0) {
```
weatherDetails.length просто

danimaxi54 commented on src/components/weatherDetails/index.tsx:
```diff
@@ -0,0 +1,115 @@
+    ];
+
+    useEffect(() => {
+        if (hourlyData && Object.keys(hourlyData).length > 0) {
```
Object.keys(hourlyData).length

danimaxi54 commented on src/components/weatherDetails/index.tsx:
```diff
@@ -0,0 +1,115 @@
+        },
+        {
+            label: 'Chance Of Rain',
+            value: humidity !== undefined ? `${humidity}%` : skeletonStyle,
```
ты должен компонент большой показывать <SkeletonLayout />
а не в каждом элементе скелетон свой показывать. но ок, пусть уже будет так

danimaxi54 commented on src/components/weatherDetails/index.tsx:
```diff
@@ -0,0 +1,115 @@
+    const [isLoading, setIsLoading] = useState(true);
+
+    useEffect(() => {
+        if (hourlyData && Object.keys(hourlyData).length > 0) {
```
Object.keys(hourlyData).length
у тебя снизу еще useEffect написал, который в deps имеет hourlyData. Объедини свои useEffect

danimaxi54 commented on src/components/weatherDetails/index.tsx:
```diff
@@ -0,0 +1,115 @@
+import FeelsLikeIcon from '../../icons/fluent_temperature-16-filled.svg'
+import WaterDropIcon from '../../icons/material-symbols_water-drop.svg'
+import UvIndexIcon from '../../icons/mingcute_sun-fill.svg'
+import {Icon} from "../svg/SvgLoader";
```
SvgLoader => Icon

danimaxi54 commented on src/components/searchCityWeather/index.tsx:
```diff
@@ -0,0 +1,94 @@
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const inputValue = e.target.value;
+        setCity(inputValue);
+        debouncedFetchCities(inputValue); // Запуск дебаунс функции
```
не нужны комменты

danimaxi54 commented on src/components/searchCityWeather/index.tsx:
```diff
@@ -0,0 +1,94 @@
+    };
+
+    const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (e.key === "Enter" && city.trim() !== '') {
```
city.trim() просто

danimaxi54 commented on src/components/searchCityWeather/index.tsx:
```diff
@@ -0,0 +1,94 @@
+
+    const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (e.key === "Enter" && city.trim() !== '') {
+            if (suggestions.length > 0) {
```
suggestions.length

danimaxi54 commented on src/components/searchCityWeather/index.tsx:
```diff
@@ -0,0 +1,94 @@
+    const debouncedFetchCities = debounce(async (query: string) => {
+        if (query.trim() !== '') {
+            try {
+                const response = await axios.get('https://autocomplete.travelpayouts.com/places2', {
```
в env переменную урл

danimaxi54 commented on src/components/searchCityWeather/index.tsx:
```diff
@@ -0,0 +1,94 @@
+    const weather = useTypedSelector((state) => state.weather);
+
+    const debouncedFetchCities = debounce(async (query: string) => {
+        if (query.trim() !== '') {
```
query.trim()

danimaxi54 commented on src/components/searchCityWeather/index.tsx:
```diff
@@ -0,0 +1,94 @@
+import {debounce} from 'lodash';
+import axios from "axios";
+import {useTypedSelector} from "../../hooks/useTypedSelector";
+import {InputSearchBar, ULSearchList} from "./SearchCityWeatherStyle";
```
ULSearchList => SearchCitiesList

danimaxi54 commented on src/components/ThisDay/index.tsx:
```diff
@@ -0,0 +1,65 @@
+    if (weather.loading) {
+        return <Skeleton height="200px" width="250px"/>;
+    }
+    if (weather.loading) {
```
что-то два раза дублируется код

danimaxi54 commented on src/components/TempUnitSwitcher/index.tsx:
```diff
@@ -0,0 +1,29 @@
+    const isMetric = unit === 'metric';
+
+    const handleChange = () => {
+        dispatch(setTemperatureUnit(isMetric ? 'imperial' : 'metric'));
```
в enum такое ложи - imperial и metric

danimaxi54 commented on src/components/TempUnitSwitcher/tempUnitSwitcherStyles.ts:
```diff
@@ -0,0 +1,46 @@
+    position: absolute;
+    top: 50%;
+    left: ${(props) => (props.checked ? 'calc(100% - 1.75rem - 2px)' : '1px')};
+    height: 1.75rem; /* 7 * 0.25rem */
```
/* 7 * 0.25rem */ зачем комменты тут

danimaxi54 commented on src/components/TempUnitSwitcher/tempUnitSwitcherStyles.ts:
```diff
@@ -0,0 +1,46 @@
+export const ToggleThumb = styled.div<{ checked: boolean }>`
+    position: absolute;
+    top: 50%;
+    left: ${(props) => (props.checked ? 'calc(100% - 1.75rem - 2px)' : '1px')};
```
left: ${(props) => (props.checked ? 'calc(100% - 1.75rem - 2px)' : '1px')};
что за сложные вычисления?

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,106 @@
+import {Skeleton} from "../../skeleton/Skeleton";
+import {useTypedSelector} from "../../hooks/useTypedSelector";
+
+
```
пустая линия

OurnoFrontendDev committed:
fix v2.0

OurnoFrontendDev committed:
Fix v2.0

OurnoFrontendDev committed:
Fix v2.0

OurnoFrontendDev committed:
Fix v2.1

OurnoFrontendDev committed:
Fix v2.2

OurnoFrontendDev committed:
1.Переделал верстку

OurnoFrontendDev committed:
1.Переделал верстку
2.URL добавил в env файл
3.Убрал дублирование
4.Исправил логику
5. Упростил UseEffect
6. Переделал логику local.storage
7. Исправил нейминги
8. Убрал сложные расчеты в css
9. Убрал пустые линии

OurnoFrontendDev committed:
1. Убрал обработку undefined в хуке UseTheme

OurnoFrontendDev committed:
1. Исправил адаптив
2. Исправил Debounce
3. Исправил отображение 7 days forecast
4. Исправил переключение температуры
5. Добавил смену цвета иконок в weather details
6. сменил цвет на темный у текстов

danimaxi54 commented on src/AppStyles.ts:
```diff
@@ -0,0 +1,77 @@
+    padding: 5px;
+    width: 100vw;
+    justify-content: flex-start;
+  }
```
отбей пустой линией снизу

danimaxi54 commented on src/AppStyles.ts:
```diff
@@ -0,0 +1,77 @@
+    justify-content: flex-start;
+  }
+
+  @media screen and (width: 540px) and (height: 720px) {
```
отбей пустой линией снизу
и в других местах тоже отбей

danimaxi54 commented on src/AppStyles.ts:
```diff
@@ -0,0 +1,77 @@
+  height: 80vh;
+  width: 65vw;
+
+    @media screen and (min-width: 500px) and (max-width: 960px) {
```
поплыла табуляция

danimaxi54 commented on src/components/Days/DayStyledComponents.ts:
```diff
@@ -0,0 +1,123 @@
+
+  @media screen and (min-width: 500px) and (max-width: 799px) {
+    padding: 6px 23px;
+  }
```
отбей пустой линией снизу
и в других местах тут

danimaxi54 commented on src/components/Days/DayStyledComponents.ts:
```diff
@@ -0,0 +1,123 @@
+    margin: 0;
+  }
+
+`;
```
отбей пустой линией снизу

danimaxi54 commented on src/components/Days/DayStyledComponents.ts:
```diff
@@ -0,0 +1,123 @@
+    font-size: 13px;
+    padding: 3px 0 3px 10px;
+  }
+
```
убери пустую линию

danimaxi54 commented on src/components/Days/DayStyledComponents.ts:
```diff
@@ -0,0 +1,123 @@
+  color: gray;
+  font-size: 24px;
+  cursor: pointer;
+  z-index: 1;
```
зачем

danimaxi54 commented on src/components/Days/DayStyledComponents.ts:
```diff
@@ -0,0 +1,123 @@
+  z-index: 1;
+  border: none;
+
+  ${({ direction }) => (direction === "left" ? "left: 0;" : "right: 0;")}
```
отбей пустой линией снизу

danimaxi54 commented on src/components/Days/DayStyledComponents.ts:
```diff
@@ -0,0 +1,123 @@
+  }
+`;
+
+export const TempStyle = styled.div`
```
TemperatureContainer <= если это контейнер

danimaxi54 commented on src/components/Days/DayStyledComponents.ts:
```diff
@@ -0,0 +1,123 @@
+  color: ${({ theme }) => theme["--temp-indicator-color"]};
+`;
+
+export const DateTimeStyle = styled.div`
```
зачем style в конце добавляешь

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,104 @@
+import { useTypedSelector } from "../../hooks/useTypedSelector";
+import { WeatherIcon } from "../weatherIcon";
+
+const visibleAreaWidth = 100;
```
VISIBLE_AREA_WIDTH

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,104 @@
+  const dailyForecasts = useTypedSelector((state) => state.weather.forecasts?.hourly || []);
+  const weather = useTypedSelector((state) => state.weather);
+
+  const [scrollPosition, setScrollPosition] = useState(0);
```
отбей пустой линией снизу

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,104 @@
+  const [scrollPosition, setScrollPosition] = useState(0);
+  const loading = weather.loading;
+
+  const formatTime = (timestamp: number) => {
```
в fomatters/date.ts вынеси

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,104 @@
+    return `${hours}:${minutesStr} ${ampm}`;
+  };
+
+  const getFilteredHourlyData = () => {
```
в formatters вынеси

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,104 @@
+    return dailyForecasts.filter((hour) => hour.dt >= startOfDay && hour.dt <= endOfDay);
+  };
+
+  const filteredData = getFilteredHourlyData();
```
в useMemo

danimaxi54 replied:
filteredHourlyData

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,104 @@
+  };
+
+  const filteredData = getFilteredHourlyData();
+  filteredData.sort((a, b) => a.dt - b.dt);
```
внутри функции filteredHourlyData делай это, тут зачем это

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,104 @@
+
+  const filteredData = getFilteredHourlyData();
+  filteredData.sort((a, b) => a.dt - b.dt);
+  const maxScroll = filteredData.length * 110 - visibleAreaWidth;
```
maxScrollЧто-За-Сущность 
не пойми для чего скролл, для какой сущности

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,104 @@
+  filteredData.sort((a, b) => a.dt - b.dt);
+  const maxScroll = filteredData.length * 110 - visibleAreaWidth;
+
+  const handleScroll = (direction: "left" | "right") => {
```
handleScrollЧто-то

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,104 @@
+    }
+  };
+
+  const weatherMappingData = filteredData.map((hour, index) => (
```
daylyWeatherItems 
хотя бы так. так куда более понятно

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,104 @@
+          ◀
+        </ArrowButton>
+        <ScrollAreaContainer>
+          <ScrollArea style={{ transform: `translateX(-${scrollPosition}px)` }}>
```
style зачем здесь

danimaxi54 commented on src/components/Days/index.tsx:
```diff
@@ -0,0 +1,104 @@
+        </ArrowButton>
+        <ScrollAreaContainer>
+          <ScrollArea style={{ transform: `translateX(-${scrollPosition}px)` }}>
+            {loading ? skeletonItems : weatherMappingData}
```
daylySkeletonItems

danimaxi54 commented on src/components/IconLoader/index.tsx:
```diff
@@ -0,0 +1,37 @@
+  if (Svg) {
+    return (
+      <Svg
+        className={classNames(className)}
```
classNames либа тут не нужна

danimaxi54 commented on src/components/SearchTogglePanel/SearchTogglePanelStyles.ts:
```diff
@@ -0,0 +1,56 @@
+import styled from "styled-components";
+
+export const CityTempThemeSwitchContainer = styled.section`
```
SwitchTemperaturMetricContainer

danimaxi54 commented on src/components/SearchTogglePanel/index.tsx:
```diff
@@ -0,0 +1,22 @@
+  SearchCityWeatherContainer, ThemeTemperatureToggleContainer,
+} from "./SearchTogglePanelStyles";
+
+export const SearchTogglePanelContainer = () => {
```
SearchToggleTemperatureUnitPanel

danimaxi54 commented on src/components/TempUnitSwitcher/index.tsx:
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { useDispatch } from "react-redux";
+import { setTemperatureUnit } from "../features/weatherSlice";
+import { HiddenInput, LabelText, SwitchLabel, ToggleThumb } from "./tempUnitSwitcherStyles";
```
tempUnitSwitcherStyles <= это что за сущность вообще такая, которая куча компонентов отдает?

danimaxi54 commented on src/components/TempUnitSwitcher/index.tsx:
```diff
@@ -0,0 +1,29 @@
+
+export const TempUnitSwitcher = () => {
+  const dispatch = useDispatch();
+
```
убери пустую линию

danimaxi54 commented on src/components/TempUnitSwitcher/index.tsx:
```diff
@@ -0,0 +1,29 @@
+export const TempUnitSwitcher = () => {
+  const dispatch = useDispatch();
+
+  const unit = useTypedSelector((state) => state.weather.unit);
```
отбей пустой линией снизу

danimaxi54 commented on src/components/TempUnitSwitcher/index.tsx:
```diff
@@ -0,0 +1,29 @@
+  const dispatch = useDispatch();
+
+  const unit = useTypedSelector((state) => state.weather.unit);
+  const isMetric = unit === TemperatureUnit.celsius;
```
отбей пустой линией снизу

danimaxi54 commented on src/components/TempUnitSwitcher/tempUnitSwitcherStyles.ts:
```diff
@@ -0,0 +1,51 @@
+`;
+
+export interface LabelTextProps {
+  readonly $isActive: boolean;
```
зачем readonly

danimaxi54 commented on src/components/TempUnitSwitcher/tempUnitSwitcherStyles.ts:
```diff
@@ -0,0 +1,51 @@
+  top: 50%;
+  transform: translateY(-50%);
+  user-select: none;
+  color: ${(props) => (props.$isActive ? "#ffffff" : "#1f2937")};
```
"#ffffff" : "#1f2937"
css переменные решил не юзать?

danimaxi54 commented on src/components/ThisDay/ThisDayStyles.ts:
```diff
@@ -0,0 +1,141 @@
+  justify-content: space-between;
+  gap: 10px;
+
+    @media screen and (min-width: 900px) {
```
табуляция поплыла

danimaxi54 commented on src/components/ThisDay/ThisDayStyles.ts:
```diff
@@ -0,0 +1,141 @@
+    justify-content: center;
+    align-items: center;
+    margin-top: 2px;
+  }
```
отбей пустой линией снизу

danimaxi54 commented on src/components/ThisDay/ThisDayStyles.ts:
```diff
@@ -0,0 +1,141 @@
+
+  @media screen and (min-width: 501px) and (max-width: 1338px) {
+    width: 50%;
+  }
```
отбей пустой линией снизу

danimaxi54 commented on src/components/ThisDay/ThisDayStyles.ts:
```diff
@@ -0,0 +1,141 @@
+    justify-content: center;
+    align-items: center;
+  }
+  
```
убери пустую линию

danimaxi54 commented on src/components/ThisDay/ThisDayStyles.ts:
```diff
@@ -0,0 +1,141 @@
+  @media screen and (max-width: 585px) {
+    font-size: 25px;
+    justify-content: center;
+  }
```
отбей пустой линией снизу

danimaxi54 commented on src/components/ThisDay/ThisDayStyles.ts:
```diff
@@ -0,0 +1,141 @@
+  @media screen and (width: 1024px) and (height: 600px) {
+    font-size: 30px;
+    padding: 0;
+  }
```
отбей пустой линией снизу и в других местах тоже

danimaxi54 commented on src/components/ThisDay/index.tsx:
```diff
@@ -0,0 +1,80 @@
+  const cityName = useTypedSelector((state) => state.weather.cityName);
+
+  const loading = weather.loading;
+
```
убери пустую линию

danimaxi54 commented on src/components/ThisDay/index.tsx:
```diff
@@ -0,0 +1,80 @@
+    dispatch(
+      fetchWeatherBySearchRequest({
+        name: cityName,
+        location: { latitude: 55.7558, longitude: 37.6173 },
```
55.7558 и 37.6173 нужно вынести в константу, чтобы разраб понимал, что за цифры это такие, к какому городу относится

danimaxi54 commented on src/components/ThisDay/index.tsx:
```diff
@@ -0,0 +1,80 @@
+    );
+  }, [dispatch]);
+
+  if (!weather.currentWeather || loading) {
```
if (loading) <= а зачем здесь weather.currentWeather еще??

danimaxi54 commented on src/components/ThisDay/index.tsx:
```diff
@@ -0,0 +1,80 @@
+    );
+  }
+
+  const todayWeather = weather.currentWeather.daily[0];
```
к верху компоненты вынеси, константы сверху находятся

danimaxi54 commented on src/components/ThisDay/index.tsx:
```diff
@@ -0,0 +1,80 @@
+    <LocationTemperatureContainer>
+      <LocationDescriptionContainer>
+        <InfoDate>{formattedDate}</InfoDate>
+        {/*<City>{city}</City>*/}
```
убирай

danimaxi54 commented on src/components/ThisDay/index.tsx:
```diff
@@ -0,0 +1,80 @@
+        </Temperature>
+      </LocationDescriptionContainer>
+      <TemperatureIconContainer>
+        {weatherInfo ? <IconLoader src={weatherInfo.image} alt={weatherInfo.description} /> : null}
```
`weatherInfo && <IconLoader src={weatherInfo.image} alt={weatherInfo.description} />`

danimaxi54 commented on src/components/WeatherUnionInfo/WeatherUnionInfoStyles.ts:
```diff
@@ -0,0 +1,107 @@
+import styled from "styled-components";
+
+export const WeatherUnionContainer = styled.section`
```
не пойми что за компонент такой

danimaxi54 commented on src/components/WeatherUnionInfo/WeatherUnionInfoStyles.ts:
```diff
@@ -0,0 +1,107 @@
+
+  @media screen and (min-width: 300px) and (max-width: 500px) {
+    padding: 20px 0 0;
+  }
```
отбей пустой линией снизу

danimaxi54 commented on src/components/WeatherUnionInfo/WeatherUnionInfoStyles.ts:
```diff
@@ -0,0 +1,107 @@
+  width: 100%;
+  gap: 24px;
+
+    @media screen and (min-width: 500px) and (max-width: 570px) {
```
табуляция поплыла

danimaxi54 commented on src/components/WeatherUnionInfo/WeatherUnionInfoStyles.ts:
```diff
@@ -0,0 +1,107 @@
+    padding: 18px 0 0;
+  }
+`;
+export const WeatherUnionInfoContainer = styled.section`
```
не пойми что за сущность

danimaxi54 commented on src/components/WeatherUnionInfo/WeatherUnionInfoStyles.ts:
```diff
@@ -0,0 +1,107 @@
+  @media screen and (min-width: 400px) and (max-width: 500px) {
+    flex-direction: column;
+    gap: 10px;
+  }
```
отбей пустой линией снизу и в других местах

danimaxi54 commented on src/components/WeatherUnionInfo/WeatherUnionInfoStyles.ts:
```diff
@@ -0,0 +1,107 @@
+  @media screen and (min-width: 300px) and (max-width: 445px) {
+    gap: 20px;
+  }
+`;
```
отбей пустой линией снизу

danimaxi54 commented on src/components/WeatherUnionInfo/WeatherUnionInfoStyles.ts:
```diff
@@ -0,0 +1,107 @@
+  width: 10%;
+  height: 100%;
+  gap: 20px;
+  flex: 1;
```
отбей пустой линией снизу и в других местах

danimaxi54 commented on src/components/WeatherUnionInfo/WeatherUnionInfoStyles.ts:
```diff
@@ -0,0 +1,107 @@
+  @media screen and (min-width: 500px) and (max-width: 1062px) {
+    min-width: 0;
+    width: 50%;
+  }
```
отбей пустой линией снизу

danimaxi54 commented on src/components/WeatherUnionInfo/WeatherUnionInfoStyles.ts:
```diff
@@ -0,0 +1,107 @@
+    width: 100%;
+    height: 100%;
+  }
+`;
```
отбей пустой линией снизу

danimaxi54 commented on src/components/WeatherUnionInfo/index.tsx:
```diff
@@ -0,0 +1,31 @@
+} from "./WeatherUnionInfoStyles";
+
+export const WeatherUnionInfo = () => {
+  return (
```
return не нужен тут

danimaxi54 commented on src/components/WeatherUnionInfo/index.tsx:
```diff
@@ -0,0 +1,31 @@
+  WeekForecastContainer,
+} from "./WeatherUnionInfoStyles";
+
+export const WeatherUnionInfo = () => {
```
не пойми название сущности

danimaxi54 commented on src/components/features/weatherSaga.ts:
```diff
@@ -0,0 +1,122 @@
+    const cityResponse: AxiosResponse<cityResponseType> = yield call(
+      axios.get,
+      `${BASE_URL_SEARCH_TOWN}`,
+      {
```
в константу вынеси параметры 
const queryParams = {}

danimaxi54 commented on src/components/features/weatherSaga.ts:
```diff
@@ -0,0 +1,122 @@
+        params: {
+          latitude,
+          longitude,
+          current: "temperature_2m,weather_code,wind_speed_10m,wind_direction_10m",
```
в константу строку

danimaxi54 commented on src/components/features/weatherSaga.ts:
```diff
@@ -0,0 +1,122 @@
+          longitude,
+          current: "temperature_2m,weather_code,wind_speed_10m,wind_direction_10m",
+          hourly:
+            "temperature_2m,precipitation_probability,wind_speed_10m,apparent_temperature,weather_code",
```
в константу строку

danimaxi54 commented on src/components/features/weatherSaga.ts:
```diff
@@ -0,0 +1,122 @@
+          hourly:
+            "temperature_2m,precipitation_probability,wind_speed_10m,apparent_temperature,weather_code",
+          daily:
+            "sunrise,sunset,precipitation_probability_max,uv_index_max,temperature_2m_max,temperature_2m_min,weather_code",
```
в константу строку

danimaxi54 commented on src/components/features/weatherSaga.ts:
```diff
@@ -0,0 +1,122 @@
+      },
+    );
+
+    const transformedData = {
```
processedWeather

danimaxi54 commented on src/components/features/weatherSaga.ts:
```diff
@@ -0,0 +1,122 @@
+        precipitation_probability: weatherResponse.data.daily.precipitation_probability_max[index],
+        weather_code: weatherResponse.data.daily.weather_code[index],
+      })),
+    };
```
отбей пустой линией снизу

danimaxi54 commented on src/components/features/weatherSaga.ts:
```diff
@@ -0,0 +1,122 @@
+  yield takeLatest(fetchWeatherBySearchRequest.type, fetchWeatherData);
+}
+
+function* handleTemperatureUnitChange() {
```
handleTemperatureUnitWorker

danimaxi54 commented on src/components/features/weatherSaga.ts:
```diff
@@ -0,0 +1,122 @@
+  const { cityName, cord } = yield select((state: RootState) => state.weather.currentWeather || {});
+  const weatherUnit: TemperatureUnit = yield select((state: RootState) => state.weather.unit);
+
+  if (cord && cityName) {
```
зачем тут if

danimaxi54 commented on src/components/features/weatherSaga.ts:
```diff
@@ -0,0 +1,122 @@
+  }
+}
+
+export function* watchTemperatureUnitChange() {
```
watchTemperatureUnitWatcher

danimaxi54 commented on src/components/features/weatherSlice.ts:
```diff
@@ -0,0 +1,53 @@
+  name: "weather",
+  initialState,
+  reducers: {
+    fetchWeatherBySearchRequest: (state, action: PayloadAction) => {
```
fetchWeatherBySearch

danimaxi54 replied:
action: PayloadAction) не используется

danimaxi54 commented on src/components/searchCityWeather/SearchCityWeatherStyle.ts:
```diff
@@ -0,0 +1,58 @@
+import styled from "styled-components";
+import { fonts } from "../../fonts/fontStyles";
+
+export const InputSearchBar = styled.input`
```
SearchWeatherInput

danimaxi54 commented on src/components/searchCityWeather/SearchCityWeatherStyle.ts:
```diff
@@ -0,0 +1,58 @@
+  @media screen and (min-width: 320px) and (max-width: 599px) {
+    padding-left: 10px;
+  }
+`;
```
отбей пустой линией снизу

danimaxi54 commented on src/components/searchCityWeather/SearchCityWeatherStyle.ts:
```diff
@@ -0,0 +1,58 @@
+export const SearchCitiesListTownItem = styled.li`
+    cursor: pointer;
+    padding: 10px;
+    
```
лишняя табуляция как будто бы

danimaxi54 commented on src/components/searchCityWeather/index.tsx:
```diff
@@ -0,0 +1,144 @@
+  const [suggestions, setSuggestions] = useState<City[]>([]);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  const debouncedCity = useDebounce(city, 500);
```
debouncedSearchWeatherCity

danimaxi54 commented on src/components/searchCityWeather/index.tsx:
```diff
@@ -0,0 +1,144 @@
+
+  const debouncedCity = useDebounce(city, 500);
+
+  const dispatch = useDispatch();
```
наверх выноси это

danimaxi54 commented on src/components/searchCityWeather/index.tsx:
```diff
@@ -0,0 +1,144 @@
+  const fetchCitiesFromNewApi = async (query: string) => {
+    if (query.trim()) {
+      try {
+        const response = await axios.get("https://geocoding-api.open-meteo.com/v1/search", {
```
почему это здесь, а не в саге??

danimaxi54 commented on src/components/searchCityWeather/index.tsx:
```diff
@@ -0,0 +1,144 @@
+  const dispatch = useDispatch();
+  const weather = useTypedSelector((state) => state.weather);
+
+  const fetchCitiesFromNewApi = async (query: string) => {
```
нужно обернуть эту функцию в useDebouncedCallback

хук useDebounce тут не поможет, так как он имеет реализацию через useEffect, а тебе нужно debounce именно функции делать

danimaxi54 commented on src/components/searchCityWeather/index.tsx:
```diff
@@ -0,0 +1,144 @@
+        handleSelectCity(suggestions[0]);
+      } else {
+        try {
+          const response = await axios.get("https://geocoding-api.open-meteo.com/v1/search", {
```
почему здесь а не в саге?

danimaxi54 commented on src/hooks/useWeatherIconInfo.tsx:
```diff
@@ -0,0 +1,54 @@
+  { code: 99, description: "Thunderstorm with heavy hail", image: "path/to/thunderstorm_with_heavy_hail.png", },
+];
+
+
```
лишняя пустая линия

danimaxi54 commented on src/hooks/useWeatherIconInfo.tsx:
```diff
@@ -0,0 +1,54 @@
+  const [weatherInfo, setWeatherInfo] = useState<WeatherInfo | null>(null);
+
+  useEffect(() => {
+    const weather = weatherCodes.find((w) => w.code === code);
```
отбей пустой линией снизу

danimaxi54 commented on src/hooks/useWeatherIconInfo.tsx:
```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from "react";
+
+const weatherCodes = [
```
WEATHER_INFO_BY_CODE_MAP

danimaxi54 commented on src/components/weekForecast/weekForecastStyles.ts:
```diff
@@ -0,0 +1,235 @@
+        padding-left: 0;
+    }
+}
+`;
```
отбей пустой линией снизу

danimaxi54 commented on src/components/weekForecast/weekForecastStyles.ts:
```diff
@@ -0,0 +1,235 @@
+    justify-content: center;
+    font-size: 12px;
+  }
+`;
```
отбей пустой линией снизу
и в других местах в этом файле

danimaxi54 commented on src/components/weatherDetails/index.tsx:
```diff
@@ -0,0 +1,128 @@
+
+  const skeletonStyle = <Skeleton />;
+
+  const weatherDetails = [
```
weatherDetailsItems

danimaxi54 commented on src/components/weatherDetails/index.tsx:
```diff
@@ -0,0 +1,128 @@
+  const uvIndex = daily[0]?.uv_index;
+  const feelsLike = hourly[0]?.apparent_temp;
+
+  const formatTime = (timestamp: number): string => {
```
в formatters/date.ts

danimaxi54 commented on src/components/weatherDetails/WeatherDetailsStyles.ts:
```diff
@@ -0,0 +1,249 @@
+        gap:5px;
+        padding-bottom: 8px;
+    }
+`;
```
отбей пустой линией снизу 
и в других местах

danimaxi54 commented on src/components/toggleTheme/toggleThemeStyles.ts:
```diff
@@ -0,0 +1,52 @@
+  $position: "left" | "right";
+}
+
+export const LabelImageThemeToggle = styled.span<LabelImageThemeToggleProps>`
```
плохое название 
WeatherToggleThemeImage

danimaxi54 commented on src/components/toggleTheme/toggleThemeStyles.ts:
```diff
@@ -0,0 +1,52 @@
+  background-color: #e0e7ff;
+`;
+
+export const HiddenInputThemeToggle = styled.input`
```
HiddenCheckboxThemeToggle

OurnoFrontendDev committed:
1. Добавил пробелы
2. Переделал хук UseDebounce
3. Вынес запров в Сага
4. Вынес функцию конвертации времени в utils
5. Поправил типизацию
6. Исправил неймиг компонентов

OurnoFrontendDev committed:
1. Добавил пробелы
2. Переделал хук UseDebounce
3. Вынес запров в Сага
4. Вынес функцию конвертации времени в utils
5. Поправил типизацию
6. Исправил неймиг компонентов

OurnoFrontendDev committed:
1. Добавил пробелы 2. Переделал хук UseDebounce 3. Вынес запров в Сага 4. Вынес функцию конвертации времени в utils 5. Поправил типизацию 6. Исправил неймиг компонентов
